lib/reply.js

```diff
@@ -148,14 +148,22 @@ Reply.prototype.send = function (payload) {
       return this
     }
 
-    if (Buffer.isBuffer(payload)) {
+    if (payload instanceof Uint8Array) {
       if (hasContentType === false) {
         this[kReplyHeaders]['content-type'] = CONTENT_TYPE.OCTET
       }
       onSendHook(this, payload)
       return this
     }
 
+    if (payload.buffer instanceof ArrayBuffer) {
+      if (hasContentType === false) {
+        this[kReplyHeaders]['content-type'] = CONTENT_TYPE.OCTET
+      }
+      onSendHook(this, Buffer.from(payload.buffer))
```

Comment:
I would expect this to work. does not?

```suggestion
      onSendHook(this, payload.buffer)
```

Reply:
Passing payload.buffer directly actually works, but by testing it seems that node's `ServerResponse.write` truncate data to an Uint8Array when passing to it the ArrayBuffer of a 'bigger' TypedArray. Using Buffer.from() ensure that the new buffer will share the same allocated memory as the ArrayBuffer we pass, and in that way the data is not truncated.